### ISSUE TITLE
Adding an AssertThrow for DG Limiters during parameter parsing

### ISSUE THREAD

**Issue by naliboff** [geodynamics/aspect#4259]
As a follow-up to #3822 and #4131, I am working on a PR to add an AssertThrow when the DG limiters are used in combination with a spherical geometry.

However, this is  less straightforward than I had imagined due to the order in which parameters are parsed.

I started by adding the check into the `parse_parameters` function within `simulator/parameters.cc`, but realized there was not a way to access the geometry model information yet.

The geometry model is accessible in the function `parse_geometry_dependent_parameters`, but the necessary parameters have not been declared yet (code compiles, but fails on runtime):
```
    prm.enter_subsection ("Discretization");
    {
      AssertThrow(prm.get_bool("Use limiter for discontinuous temperature solution") == true &&
                  geometry_model.natural_coordinate_system() == Utilities::Coordinates::spherical,
                  ExcMessage("The limiter for the discontinuous temperature solution has not been tested "
                             "in a spherical geometry and currently requires the use of a cartesian "
                             "geometry model."));
      AssertThrow(prm.get_bool("Use limiter for discontinuous composition solution") == true &&
                  geometry_model.natural_coordinate_system() == Utilities::Coordinates::spherical,
                  ExcMessage("The limiter for the discontinuous composition solution has not been tested "
                             "in a spherical geometry and currently requires the use of a cartesian "
                             "geometry model."));
    }
    prm.leave_subsection ();
```

**Suggestions for where this check should go and how to proceed?** 

I suppose it go into solver.cc where the limiter function is called, but this seems like a check that should go during the parameter parsing.

Is a new function called from `solver.cc `needed for this check?

Thanks for the input!

**Comment by tjhei** [comment]
Instead of a more complicated solution, I would just put it somewhere here: https://github.com/geodynamics/aspect/blob/c751ae97528536e7d4f95a4bb2f061e34f145a8e/source/simulator/core.cc#L439

**Comment by naliboff** [comment]
@tjhei - thanks! PR is located at #4262.

**Comment by bangerth** [comment]
Fixed by #4262.
